ethtest/test_case_splitter.go

```diff
@@ -59,48 +69,74 @@ func sortForks(log logger.Logger, cfgForks []string) (forks []string) {
 
 type TestCaseSplitter struct {
 	enabledForks []string  // Which forks are enabled by user (default is all)
-	jsons        []*stJSON // Decoded json files
+	jsons        []*stJSON // Decoded json fil
 	log          logger.Logger
+	chainConfigs map[string]*params.ChainConfig
 }
 
 // SplitStateTests iterates unmarshalled Geth-State test-files and divides them by 1) fork and
 // 2) tests cases. Each file contains 1..N enabledForks, one block environment (marked as 'env') and one
 // input alloc (marked as 'env'). Each fork within a file contains 1..N tests (marked as 'post').
-func (s *TestCaseSplitter) SplitStateTests() (dividedTests []txcontext.TxContext) {
+func (s *TestCaseSplitter) SplitStateTests() (dividedTests []Transaction, err error) {
 	var overall uint32
 
 	// Iterate all JSONs
 	for _, stJson := range s.jsons {
-		number := 0
-		env := &stJson.Env
 		baseFee := stJson.Env.BaseFee
 		if baseFee == nil {
 			// ethereum uses `0x10` for genesis baseFee. Therefore, it defaults to
 			// parent - 2 : 0xa as the basefee for 'this' context.
 			baseFee = &BigInt{*big.NewInt(0x0a)}
 		}
 
-		// TODO: each test requires its own block context and chain config
 		// Iterate all usable forks within one JSON file
 		for _, fork := range s.enabledForks {
-			posts := stJson.Post[fork]
+			postNumber := 0
+			posts, ok := stJson.Post[fork]
+			if !ok {
+				continue
+			}
+			chainCfg, err := s.getChainConfig(fork)
+			if err != nil {
+				return nil, err
+			}
 			// Iterate all tests within one fork
 			for _, post := range posts {
-				number++
+				postNumber++
 				msg, err := stJson.Tx.toMessage(post, baseFee)
 				if err != nil {
-					s.log.Warningf("Path: %v, fork: %v, test number: %v\n"+
-						"cannot decode tx to message: %v", stJson.path, fork, number, err)
+					s.log.Warningf("Path: %v, fork: %v, test postNumber: %v\n"+
+						"cannot decode tx to message: %v", stJson.path, fork, postNumber, err)
 					continue
 				}
 
-				dividedTests = append(dividedTests, newStateTestTxContest(stJson, env, msg, post.RootHash, fork, number))
+				if fork == "Paris" {
+					fork = "Merge"
+				}
+				ctx := newStateTestTxContest(stJson, msg, post, chainCfg, fork, postNumber)
+				dividedTests = append(dividedTests, Transaction{
+					fork,
+					ctx,
+				})
```

Comment:
I feel `ctx` is too generic. 
```suggestion
				txCtx := newStateTestTxContest(stJson, msg, post, chainCfg, fork, postNumber)
				dividedTests = append(dividedTests, Transaction{
					fork,
					txCtx,
				})
```